### PR TITLE
Reworked for faster coding of records

### PR DESCRIPTION
**Pull request by MartinWoodONS**
CHANGES:
- refactoring of coder class (less pandas = faster)
- replaced FuzzyWuzzy with RapidFuzz
- changed fuzzy matching algo to token_set_ratio
- reimplemented parallelism using modin (optional to user)
- more tests
- tox versions updated to 3.7, 3.8, 3.9

ISSUES:
- parallelism test disabled because it doesn't play nice with testr parallelism in tox (parallel tests of parallel processes create many, many warnings!), I'd love to make this work
- due to modin requirement, doesn't install in python versions 3.5/3.6 anymore (tox updated to reflect this), have submitted because these python versions are outdated.  Can alternatively specify older pre-v1.0 modin packages.

### CODE REVIEW

**Comment by aeturrell** [comment]
Wow @MartinWoodONS this looks amazing! Probably not going to have time to look at it really soon so have asked @djyldyz to review. @djyldyz: no pressure though, let us know if you're not available for this and I'll pick it up when I can.

File: tests/test_occupationcoder.py [hunk 1]
```diff
@@ -2,43 +2,162 @@
...
+        # in same environment from which tests are called
+        subprocess.run([sys.executable, '-m',
+                        'occupationcoder.coder',
                         'tests/test_vacancies.csv'])
         df = pd.read_csv(os.path.join('occupationcoder',
                                       'outputs',
                                       'processed_jobs.csv'))
         self.assertEqual(df['SOC_code'].to_list(), [211, 242, 912])
+
+    def manual_load_test(self):
```
Comment (djyldyz) [inline tests/test_occupationcoder.py:116]: Should we add test prefix to run as part of testing? As in `test_manual_load_test`

File: occupationcoder/coder.py [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+        self.df_columns.update({"title": title_column,
+                                "sector": sector_column,
+                                "description": description_column})
+
+        # Overwrite to save memory after conversion to Modin DataFrame
+        record_df = mpd.DataFrame(record_df)
+        record_df['SOC_code'] = record_df.apply(self._code_row, axis=1)
+
+        # Hack a private method to convert back to Pandas DataFrame
+        result = record_df._to_pandas()
```
Comment (djyldyz) [inline occupationcoder/coder.py:225]: Tiny thing, I think this private method is fine, but sometimes they are not updated between versions, so could break.

File: README.rst [hunk 3]
```diff
@@ -130,8 +130,9 @@ use
 .. code-block:: python
 
     df = pd.read_csv('path/to/foo.csv')
-    df = myCoder.codedataframe(df)
+    df = myCoder.code_data_frame(df, title_column='job_title', sector_column='job_sector', description_column='job_description')
```
Comment (djyldyz) [inline README.rst:133]: Do you mind mentioning that it's possible to run the algorithm with `modin` using:
`df = myCoder.parallel_code_data_frame(df, title_column='job_title', sector_column='job_sector', description_column='job_description')`

**Review by djyldyz — APPROVED** [review]
This is in great shape. Tests run successfully (including optional ones to test loading and parallel processing). I'm also able to import and use coder interactively.

The `Coder` class is much improved: reads well and only use pandas where necessary.

**Pull request merged**